### PR TITLE
Add `query_scheduler` root block to docs

### PR DESCRIPTION
**Pull request by dimitarvdimitrov**
Signed-off-by: Dimitar Dimitrov <dimitar.dimitrov@grafana.com>

#### What this PR does

The GEM docs [have the `query_scheduler` as a root block](https://grafana.com/docs/metrics-enterprise/latest/config/reference/#query_scheduler_config) and it looks like something that should go into a root block for Mimir too, but was missed. This PR adds it.

#### Which issue(s) this PR fixes or relates to

Fixes #<issue number>

#### Checklist

- [ ] Tests updated
- [ ] Documentation added
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

**Review by 56quarters — APPROVED** [review]
LGTM, thanks!

File: tools/doc-generator/parse/root_blocks.go
Comment (pracucci) [inline tools/doc-generator/parse/root_blocks.go:65]: Editing to use the naming we agreed on when working on doc.

```suggestion
			Desc:       "The query_scheduler configures query-scheduler component.",
```

**Review by pracucci — APPROVED** [review]
LGTM (modulo a nit)

File: tools/doc-generator/parse/root_blocks.go
Comment (dimitarvdimitrov) [reply tools/doc-generator/parse/root_blocks.go]: fixed in https://github.com/grafana/mimir/pull/1623/commits/08b911c1c89331b553c9808fea170b7bf19e8678

File: tools/doc-generator/parse/root_blocks.go
Comment (pracucci) [inline tools/doc-generator/parse/root_blocks.go:65]: Sorry, I jus realized it's not consistent with the other descriptions. Let's make it consistent. I know it's a nit, but I've seen docs squad to be quite keen on that. Thanks!

```suggestion
			Desc:       "The query_scheduler block configures query-scheduler.",
```

**Review by pracucci — APPROVED** [review]

File: tools/doc-generator/parse/root_blocks.go
Comment (dimitarvdimitrov) [reply tools/doc-generator/parse/root_blocks.go]: no worries, I should have looked as well. Honestly, they feel redundant to me, but not my area of expertise.

pushed the change in 2fd049d655ab7eee30cd87cdc6e7df29b2ba5d90

Comment (pracucci) [reply tools/doc-generator/parse/root_blocks.go]: Could you rebuild the doc? Then we merge it, promised! Actually I already push the auto-merge button ;)

**Pull request merged**
